aries_cloudagent/connections/base_manager.py

```diff
@@ -665,16 +667,29 @@ def diddoc_connection_targets(
                 )
         return targets
 
-    async def fetch_did_document(self, did: str) -> Tuple[DIDDoc, StorageRecord]:
+    async def fetch_did_document(
+        self, did: str
+    ) -> Tuple[Union[DIDDoc, ResolvedDocument], StorageRecord]:
         """Retrieve a DID Document for a given DID.
 
         Args:
             did: The DID to search for
         """
-        async with self._profile.session() as session:
-            storage = session.inject(BaseStorage)
-            record = await storage.find_record(self.RECORD_TYPE_DID_DOC, {"did": did})
-        return DIDDoc.from_json(record.value), record
+        if DID.is_valid(did):
+            async with self._profile.session() as session:
+                storage = session.inject(BaseStorage)
+                record = await storage.find_record(
+                    self.RECORD_TYPE_DID_DOCUMENT, {"did": did}
+                )
+            return ResolvedDocument.from_json(record.value), record
+
+        else:  # legacy documents for unqualified dids
+            async with self._profile.session() as session:
+                storage = session.inject(BaseStorage)
+                record = await storage.find_record(
+                    self.RECORD_TYPE_DID_DOC, {"did": did}
+                )
+            return DIDDoc.from_json(record.value), record
```

Comment:
I recommend not making it a responsibility of the `BaseConnectionManager` to store and recall DID Documents for did:peer:3 DIDs. It made sense to do this for the legacy peer DID resolver but in this case I think the encapsulation is better to have the peer DID 3 resolver handle it.